linux/ibus-keyman/test/test_ibus_keyman.py

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/python3
+
+import gi
+import logging
+import os
+import sys
+import time
+from threading import Timer
+
+from pynput.keyboard import Key, Controller
+
+try:
+    from xkbgroup import XKeyboard
+    has_xkbgroup = True
+except:
+    has_xkbgroup = False
+
+gi.require_version('Gtk', '3.0')
+gi.require_version('Gdk', '3.0')
+from gi.repository import Gtk, Gdk, Gio
+gi.require_version('IBus', '1.0')
+from gi.repository import IBus
+
+from change_keyboard import change_keyboard, bus_has_engine, get_current_engine
+
+class TestView(Gtk.Window):
+
+    def __init__(self, test_name):
+        Gtk.Window.__init__(self, title=test_name)
+        self.known_modifiers = { "SHIFT" : Key.shift,
+            "LCTRL" : Key.ctrl_l,
+            "RCTRL" : Key.ctrl_r,
+            "LALT" : Key.alt_l,
+            "RALT" : Key.alt_r }
+        self.known_keys = { "K_A" : "a",
```

Comment:
Can we make this a complete set so we can add other tests later that use other keys without breaking the test harness?

Reply:
ok